core/src/test/java/org/apache/iceberg/rest/requests/TestPlanTableScanRequest.java

```diff
@@ -0,0 +1,153 @@
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid request: planTableScanRequest null");
+  }
+
+  @Test
+  public void roundTripSerdeWithEmptyObjectAndDefaultsPresent() {
+    PlanTableScanRequest request = new PlanTableScanRequest.Builder().build();
+
+    String expectedJson = "{" + "\"case-sensitive\":true," + "\"use-snapshot-schema\":false}";
+
+    String json = PlanTableScanRequestParser.toJson(request, false);
+    assertThat(json).isEqualTo(expectedJson);
+
+    // can't do an equality comparison on PlanTableScanRequest because we don't implement
+    // equals/hashcode
+    assertThat(PlanTableScanRequestParser.toJson(PlanTableScanRequestParser.fromJson(json), false))
+        .isEqualTo(expectedJson);
+  }
+
+  @Test
+  public void roundTripSerdeWithSelectField() {
+    PlanTableScanRequest request =
+        new PlanTableScanRequest.Builder()
+            .withSnapshotId(1L)
+            .withSelect(Lists.newArrayList("col1", "col2"))
+            .build();
+
+    String expectedJson =
+        "{\"snapshot-id\":1,"
+            + "\"select\":[\"col1\",\"col2\"],"
+            + "\"case-sensitive\":true,"
+            + "\"use-snapshot-schema\":false}";
+
+    String json = PlanTableScanRequestParser.toJson(request, false);
+    assertThat(json).isEqualTo(expectedJson);
+
+    // can't do an equality comparison on PlanTableScanRequest because we don't implement
+    // equals/hashcode
+    assertThat(PlanTableScanRequestParser.toJson(PlanTableScanRequestParser.fromJson(json), false))
+        .isEqualTo(expectedJson);
+  }
+
+  @Test
+  public void roundTripSerdeWithFilterField() {
+    PlanTableScanRequest request =
+        new PlanTableScanRequest.Builder()
+            .withSnapshotId(1L)
+            .withFilter(Expressions.alwaysFalse())
+            .build();
+
+    String expectedJson =
+        "{\"snapshot-id\":1,"
+            + "\"filter\":\"false\","
+            + "\"case-sensitive\":true,"
+            + "\"use-snapshot-schema\":false}";
+
+    String json = PlanTableScanRequestParser.toJson(request, false);
+    assertThat(json).isEqualTo(expectedJson);
+
+    // can't do an equality comparison on PlanTableScanRequest because we don't implement
+    // equals/hashcode
+    assertThat(PlanTableScanRequestParser.toJson(PlanTableScanRequestParser.fromJson(json), false))
+        .isEqualTo(expectedJson);
+  }
+
+  @Test
+  public void roundTripSerdeWithAllFieldsInvalidRequest() {
+    PlanTableScanRequest request =
+        new PlanTableScanRequest.Builder()
+            .withSnapshotId(1L)
+            .withSelect(Lists.newArrayList("col1", "col2"))
+            .withFilter(Expressions.alwaysTrue())
+            .withStartSnapshotId(1L)
+            .withEndSnapshotId(2L)
+            .withCaseSensitive(false)
+            .withUseSnapshotSchema(true)
+            .withStatsFields(Lists.newArrayList("col1", "col2"))
+            .build();
```

Comment:
Shouldn't building this response with both a start/end + snapshot id fail at the time of building itself? Why wait until serialization?